member_map.go

```diff
@@ -103,33 +104,56 @@ func (m *Member) GetID() MemberID {
 type MemberMap struct {
 	lock    sync.RWMutex
 	members map[MemberID]*Member
-
-	// This is for selecting k random member based on round-robin
-	waitingMembers []Member
 }
 
 func NewMemberMap() *MemberMap {
 	return &MemberMap{
-		members:        make(map[MemberID]*Member),
-		waitingMembers: make([]Member, 0),
-		lock:           sync.RWMutex{},
+		members: make(map[MemberID]*Member),
+		lock:    sync.RWMutex{},
 	}
 }
 
-// Select K random memberID from waitingMembers(length of returning member can be lower than k).
-// ** WaitingMembers are shuffled every time when members are updated **, so just returning first K item in waitingMembers is same as
-// selecting k random membersID.
+// Select K random member (length of returning member can be lower than k).
 func (m *MemberMap) SelectKRandomMemberID(k int) []Member {
-	return nil
+
+	m.lock.Lock()
+	defer m.lock.Unlock()
+
+	// Filter non-alive member
+	members := make([]Member, 0)
+	for _, member := range m.members {
+		if member.Status == Alive {
```

Comment:
Is it ok that it doesn't include the suspected members?
I think only alive members could be selected, but plz check it and reply to me :)

Reply:
I think a suspected member is a member that may be dead, so it is not appropriate to send an indirect-ping.